fml/platform/fuchsia/message_loop_fuchsia.cc

```diff
@@ -11,28 +11,52 @@
 namespace fml {
 
 MessageLoopFuchsia::MessageLoopFuchsia()
-    : loop_(&kAsyncLoopConfigAttachToCurrentThread) {}
+  : loop_(&kAsyncLoopConfigAttachToCurrentThread),
+    running_(false) {
+  auto status = zx_timer_create(ZX_TIMER_SLACK_LATE, ZX_CLOCK_MONOTONIC, &timer_);
+  FML_CHECK(status == ZX_OK);
+  timer_wait_ = new async::Wait(timer_, ZX_TIMER_SIGNALED);
+
+  auto wait_handler = [&](async_dispatcher_t* dispatcher,
+                        async::Wait* wait,
+                        zx_status_t status,
+                        const zx_packet_signal_t* signal
+                    ) {
+    RunExpiredTasksNow();
+    wait->Begin(loop_.dispatcher());
```

Comment:
Before beginning the new wait, the `ZX_TIMER_SIGNALED` signal must be reset manually via `zx_timer_cancel`. Otherwise, it will only be reset on the next call to `zx_timer_set` which may not happen immediately (or at all). Till the next `zx_timer_set` you will be stuck in a busy loop waiting on a signal.

Reply:
A gotcha to watch out for: Make sure to call `zx_timer_cancel` before `RunExpiredTasksNow` lest you risk cancelling potential timer sets in the `RunExpiredTasksNow`.